builder/src/main/java/dev/sashimono/builder/jar/SignatureTask.java

```diff
@@ -0,0 +1,44 @@
+package dev.sashimono.builder.jar;
+
+import static java.lang.Boolean.getBoolean;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.Function;
+
+import dev.sashimono.builder.util.TaskMap;
+
+public class SignatureTask implements Function<TaskMap, Void> {
+
+    @Override
+    public Void apply(final TaskMap taskMap) {
+        if (getBoolean("sign_artifacts")) {
+            final Path executablePath = Path.of(System.getenv("GPG_EXECUTABLE_PATH"));
+            final String keyName = System.getenv("GPG_KEYNAME");
+            final String passPhrase = System.getenv("GPG_PASSPHRASE");
+
+            final List<FileOutput> outputs = taskMap.results(FileOutput.class);
+            for (final FileOutput i : outputs) {
+                final Path inputPath = i.file().toAbsolutePath();
+                final String inputFileName = inputPath.getFileName().toString();
+                if (inputFileName.endsWith(".jar") || inputFileName.endsWith(".pom")) {
+                    final Path outputPath = inputPath
+                            .resolveSibling(inputFileName + ".asc");
+                    final ProcessBuilder processBuilder = new ProcessBuilder();
+                    final List<String> args = List.of(executablePath.toString(), "--local-user", keyName, "--passphrase",
+                            passPhrase, "--batch", "--no-tty", "--armor", "--detach-sign", "--pinentry-mode", "loopback",
+                            "--output", outputPath.toString(), "--sign", inputPath.toString());
+                    processBuilder.command(args);
+                    try {
+                        processBuilder.start();
```

Comment:
You need to wait for this to complete here, and you also need to consume the output (stderr and stdout) of the process to prevent hangs. If you don't read from the streams it might work as long as the output is small enough to buffer, but if it produces a bit too much output it will hang.

Reply:
Is it not synchronous?

You're right I didn't need to read from the streams but I'd also enabled batch and no-tty so is that still a factor?